sdk/couchbase-core/src/mgmtx/bucket_settings.rs

```diff
@@ -0,0 +1,515 @@
+        match &self.0 {
+            InnerCompressionMode::Off => write!(f, "off"),
+            InnerCompressionMode::Passive => write!(f, "passive"),
+            InnerCompressionMode::Active => write!(f, "active"),
+            InnerCompressionMode::Other(val) => write!(f, "unknown({})", val),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub(crate) enum InnerCompressionMode {
+    Off,
+    Passive,
+    Active,
+    Other(String),
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub struct DurabilityLevel(InnerDurabilityLevel);
+
+impl DurabilityLevel {
+    pub const NONE: DurabilityLevel = DurabilityLevel(InnerDurabilityLevel::None);
+
+    pub const MAJORITY: DurabilityLevel = DurabilityLevel(InnerDurabilityLevel::Majority);
+
+    pub const MAJORITY_AND_PERSIST_ACTIVE: DurabilityLevel =
+        DurabilityLevel(InnerDurabilityLevel::MajorityAndPersistActive);
+
+    pub const PERSIST_TO_MAJORITY: DurabilityLevel =
+        DurabilityLevel(InnerDurabilityLevel::PersistToMajority);
+
+    pub(crate) fn other(val: String) -> DurabilityLevel {
+        DurabilityLevel(InnerDurabilityLevel::Other(val))
+    }
+}
+
+impl Display for DurabilityLevel {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match &self.0 {
+            InnerDurabilityLevel::None => write!(f, "none"),
+            InnerDurabilityLevel::Majority => write!(f, "majority"),
+            InnerDurabilityLevel::MajorityAndPersistActive => write!(f, "majorityAndPersistActive"),
+            InnerDurabilityLevel::PersistToMajority => write!(f, "persistToMajority"),
+            InnerDurabilityLevel::Other(val) => write!(f, "unknown({})", val),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub(crate) enum InnerDurabilityLevel {
+    None,
+    Majority,
+    MajorityAndPersistActive,
+    PersistToMajority,
+    Other(String),
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub struct ConflictResolutionType(InnerConflictResolutionType);
+
+impl ConflictResolutionType {
+    pub const SEQUENCE_NUMBER: ConflictResolutionType =
+        ConflictResolutionType(InnerConflictResolutionType::SequenceNumber);
+
+    pub const TIMESTAMP: ConflictResolutionType =
+        ConflictResolutionType(InnerConflictResolutionType::Timestamp);
+
+    pub const CUSTOM: ConflictResolutionType =
+        ConflictResolutionType(InnerConflictResolutionType::Custom);
+
+    pub(crate) fn other(val: String) -> ConflictResolutionType {
+        ConflictResolutionType(InnerConflictResolutionType::Other(val))
+    }
+}
+
+impl Display for ConflictResolutionType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match &self.0 {
+            InnerConflictResolutionType::SequenceNumber => write!(f, "seqno"),
+            InnerConflictResolutionType::Timestamp => write!(f, "lww"),
```

Comment:
Why is this lww?

Reply:
That I do not know, "last write wins" maybe?

Reply:
That would make sense, thanks.

Reply:
Yeah, last write wins is what it stands for